src/Nethermind/Nethermind.Optimism/InitializeBlockchainOptimism.cs

```diff
@@ -18,94 +18,80 @@
     {
-        if (_api.SpecProvider is null) throw new StepDependencyException(nameof(_api.SpecProvider));
-        if (_api.SpecHelper is null) throw new StepDependencyException(nameof(_api.SpecHelper));
-        if (_api.L1CostHelper is null) throw new StepDependencyException(nameof(_api.L1CostHelper));
-        if (_api.WorldState is null) throw new StepDependencyException(nameof(_api.WorldState));
+        if (api.SpecProvider is null) throw new StepDependencyException(nameof(api.SpecProvider));
+        if (api.SpecHelper is null) throw new StepDependencyException(nameof(api.SpecHelper));
+        if (api.L1CostHelper is null) throw new StepDependencyException(nameof(api.L1CostHelper));
+        if (api.WorldState is null) throw new StepDependencyException(nameof(api.WorldState));
 
         return new OptimismTransactionProcessor(
-            _api.SpecProvider,
-            _api.WorldState,
+            api.SpecProvider,
+            api.WorldState,
             virtualMachine,
-            _api.LogManager,
-            _api.L1CostHelper,
-            _api.SpecHelper,
+            api.LogManager,
+            api.L1CostHelper,
+            api.SpecHelper,
             codeInfoRepository
         );
     }
 
     protected override IHeaderValidator CreateHeaderValidator()
     {
-        if (_api.InvalidChainTracker is null) throw new StepDependencyException(nameof(_api.InvalidChainTracker));
+        if (api.InvalidChainTracker is null) throw new StepDependencyException(nameof(api.InvalidChainTracker));
 
         OptimismHeaderValidator opHeaderValidator = new(
-            _api.BlockTree,
-            _api.SealValidator,
-            _api.SpecProvider,
-            _api.LogManager);
+            api.BlockTree,
+            api.SealValidator,
+            api.SpecProvider,
+            api.LogManager);
 
-        return new InvalidHeaderInterceptor(opHeaderValidator, _api.InvalidChainTracker, _api.LogManager);
+        return new InvalidHeaderInterceptor(opHeaderValidator, api.InvalidChainTracker, api.LogManager);
     }
 
     protected override IBlockValidator CreateBlockValidator()
     {
-        if (_api.InvalidChainTracker is null) throw new StepDependencyException(nameof(_api.InvalidChainTracker));
-        if (_api.TxValidator is null) throw new StepDependencyException(nameof(_api.TxValidator));
-
-        BlockValidator blockValidator = new(
-            _api.TxValidator.WithValidator(TxType.DepositTx, new OptimismTxValidator()),
-            _api.HeaderValidator,
-            _api.UnclesValidator,
-            _api.SpecProvider,
-            _api.LogManager);
-
-        return new InvalidBlockInterceptor(blockValidator, _api.InvalidChainTracker, _api.LogManager);
+        if (api.InvalidChainTracker is null) throw new StepDependencyException(nameof(api.InvalidChainTracker));
+        return new InvalidBlockInterceptor(base.CreateBlockValidator(), api.InvalidChainTracker, api.LogManager);
```

Comment:
Is this needed? I don't think we need to create interceptor here at all?

Reply:
@LukaszRozmej we need to not invalidate blocks because of the deposit tx. So we need a block validator that uses our tx validator.

Note: Optimism DepositTxs are not like Aura SystemTxs. DepositTxs are actually included in blocks.

Reply:
I am confused. With my implementation it will use TxValidator, this validator will go to per tx type validators. I register Deposit validator earlier in this file. So deposit transactions won't be validated, but other types will. Is that bad?

Reply:
my bad. didn't notice the registration of deposit tx validator on line 27.